src/main/java/com/kodilla/ecommercee/service/UserService.java

```diff
@@ -0,0 +1,53 @@
+package com.kodilla.ecommercee.service;
+
+import com.kodilla.ecommercee.domain.User;
+import com.kodilla.ecommercee.domain.UserKey;
+import com.kodilla.ecommercee.dto.UserDto;
+import com.kodilla.ecommercee.dto.UserKeyDto;
+import com.kodilla.ecommercee.exception.CreatingObjectWithIdException;
+import com.kodilla.ecommercee.exception.EntityNotFoundException;
+import com.kodilla.ecommercee.mapper.UserKeyMapper;
+import com.kodilla.ecommercee.mapper.UserMapper;
+import com.kodilla.ecommercee.repository.UserRepository;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Random;
+
+@AllArgsConstructor
+@Service
+public class UserService {
+
+    private final UserMapper userMapper;
+    private final UserKeyMapper userKeyMapper;
+    private final UserRepository repository;
+
+    @Transactional
+    public UserDto create(final UserDto userDto) throws CreatingObjectWithIdException {
+        if(userDto.getId() != 0) throw new CreatingObjectWithIdException(User.class, userDto.getId());
+        return saveAndMapToDto(userDto);
+    }
+
+    private UserDto saveAndMapToDto(final UserDto userDto) {
+        User user = repository.save(userMapper.toEntity(userDto));
+        return userMapper.toDto(user);
+    }
+
+    @Transactional
+    public void block(final long id) throws EntityNotFoundException {
+        User user = repository.findById(id).orElseThrow(() -> new EntityNotFoundException(User.class, id));
+        repository.save(new User(user.getId(),user.getUsername(), false, user.getUserKey(), user.getCarts(), user.getOrders()));
+    }
+
+    @Transactional
+    public UserKeyDto generateKey(final long id) throws EntityNotFoundException {
+        User user = repository.findById(id).orElseThrow(() -> new EntityNotFoundException(User.class, id));
+        Random random = new Random();
+        UserKey key =  new UserKey(random.nextLong(), Instant.now().plus(Duration.ofHours(1)));
+        repository.save(new User(user.getId(),user.getUsername(), user.isStatus(), key, user.getCarts(), user.getOrders()));
```

Comment:
ja bym tutaj rezultat tej metody przypisał do zmiennej `user`...

Reply:
czemu jest new User? Chyba powinnismy zmodyfikowac poprzedniego uzytkownika, a nie tworzyc nowego?

Reply:
Skoro nie używamy setterów aby zachować niemutowalność obiektów to trzeba w tym wypadku utworzyć nowy obiekt na podstawie pól poprzedniego ale podstawiając nowy klucz

Reply:
Niestety przy Hibernate jest problem z niemutowalnoscia i w przypadku tej encji dopuscilbym mutiwanie encji. Jest to lepsze niz tworzenie nowego obiektu (tracimy tozsamosc obiektu).

Reply:
W takim razie dodam `@Setter` do User